nixos/modules/services/monitoring/prometheus/exporters/fronius.nix

```diff
@@ -0,0 +1,79 @@
+{ config, lib, pkgs, options }:
+
+with lib;
+
+let
+  cfg = config.services.prometheus.exporters.fronius;
+in
+{
+  port = 8080;
+  extraOpts = {
+    symo = {
+      url = mkOption {
+        type = types.str;
+        description = lib.mdDoc ''
+          Target base URL of Fronius Symo device. (default "http://symo.ip.or.hostname")
+        '';
+      };
+      enable-archive = mkOption {
+        type = types.nullOr types.bool;
+        description = lib.mdDoc ''
+          Enable/disable scraping of archive data (default true)
+        '';
+        default = true;
+      };
+      enable-power-flow = mkOption {
+        type = types.nullOr types.bool;
+        description = lib.mdDoc ''
+          Enable/disable scraping of power flow data (default true)
+        '';
+        default = true;
+      };
+      header = mkOption {
+        type = types.nullOr types.str;
+        description = lib.mdDoc ''
+          List of "key: value" headers to append to the requests going to Fronius Symo. Example: --symo.header "authorization=Basic <base64>".
+        '';
+        default = null;
+      };
+      timeout = mkOption {
+        type = types.nullOr types.str;
+        description = lib.mdDoc ''
+          Timeout in seconds when collecting metrics from Fronius Symo. Should not be larger than the scrape interval. (default 5)
+        '';
+        default = "5s";
+      };
+    };
+    log.level = mkOption {
+      type = types.nullOr types.str;
+      description = lib.mdDoc ''
+        Logging level. (default "info")
+      '';
+      default = "info";
+    };
+  };
+  serviceOpts = {
+    serviceConfig = {
+      Environment = [
+        ## IP Address to bind to listen for Prometheus scrapes.
+        "BIND_ADDR=${cfg.listenAddress}:${toString cfg.port}"
+        ## Target URL of Fronius Symo device.
+        "SYMO__URL=${cfg.symo.url}"
+      ]
+        ## Enable/disable scraping of archive data
+        ++ lib.optional (cfg.symo.enable-archive != null) "SYMO__ENABLE_ARCHIVE=${builtins.toString cfg.symo.enable-archive}"
+        ## Enable/disable scraping of power flow data
+        ++ lib.optional (cfg.symo.enable-power-flow != null) "SYMO__ENABLE_POWER_FLOW=${builtins.toString cfg.symo.enable-power-flow}"
```

Comment:
This usage of `toString <bool>` will result in `SYMO__ENABLE_ARCHIVE=1` and `SYMO_ENABLE_ARCHIVE=` instead of the (perhaps expected) `...=true`/`...=false` strings. [The example](https://github.com/ccremer/fronius-exporter/blob/58c831bfda33efd26b7df7bdbd2153fe30c6b30e/examples/systemd.env#L8) uses true and false, so it would make sense to test if `1` and the empty string result in the same functionality, or (probably better) just map to `true` and `false` instead.